externs/browser/html5.js

```diff
@@ -4286,3 +4288,102 @@ Plugin.prototype.length;
 
 /** @type {string} */
 Plugin.prototype.name;
+
+/** @record */
+function CustomElementInterface() {}
+
+/** @type {!Array<string>|undefined} */
+CustomElementInterface.observedAttributes;
+
+/** @type {function()|undefined} */
+CustomElementInterface.prototype.connectedCallback;
+
+/** @type {function()|undefined} */
+CustomElementInterface.prototype.disconnectedCallback;
+
+/** @type {function(string, ?string, ?string, ?string)|undefined} */
+CustomElementInterface.prototype.attributeChangedCallback;
+
+/**
+ * @constructor
+ * @extends {HTMLElement}
+ * @implements {CustomElementInterface}
+ */
+function CustomElement () {}
+
+/** @type {!Array<string>|undefined} */
+CustomElement.observedAttributes;
+
+/** @type {function()|undefined} */
+CustomElement.prototype.connectedCallback;
+
+/** @type {function()|undefined} */
+CustomElement.prototype.disconnectedCallback;
+
+/** @type {function(string, ?string, ?string, ?string)|undefined} */
+CustomElement.prototype.attributeChangedCallback;
+
+/**
+ * @see https://w3c.github.io/webcomponents/spec/custom/#custom-elements-api
+ * @const
+ */
+var customElements = {};
```

Comment:
I think this should be of type `CustomElementRegistry`, which is also constructable.